keps/sig-scheduling/2458-preferred-fit-strategy/README.md

```diff
@@ -0,0 +1,664 @@
+    // Points defining priority function shape
+    Shape []UtilizationShapePoint
+}
+
+// Note that the two types defined below already exist in the scheduler's component config API.
+// ResourceSpec represents a single resource.
+type ResourceSpec struct {
+	// Name of the resource.
+	Name string
+	// Weight of the resource.
+	Weight int64
+}
+
+// UtilizationShapePoint represents a single point of a priority function shape.
+type UtilizationShapePoint struct {
+	// Utilization (x axis). Valid values are 0 to 100. Fully utilized node maps to 100.
+	Utilization int32
+	// Score assigned to a given utilization (y axis). Valid values are 0 to 10.
+	Score int32
+}
+
+```
+
+Add `PreferredFitStrategy` to `NodeResourcesFitArgs`:
+
+```go
+
+// NodeResourcesFitArgs holds arguments used to configure the NodeResourcesFit plugin.
+type NodeResourcesFitArgs struct {
+	metav1.TypeMeta
+
+	// IgnoredResources is the list of resources that NodeResources fit filter
+	// should ignore.
+	IgnoredResources []string
+	// IgnoredResourceGroups defines the list of resource groups that NodeResources fit 
+    // filter should ignore.
+	// e.g. if group is ["example.com"], it will ignore all resource names that begin
+	// with "example.com", such as "example.com/aaa" and "example.com/bbb".
+	// A resource group name can't contain '/'.
+	IgnoredResourceGroups []string
+    
+    // PreferredFitStrategy selects the node resource scoring strategy.
+    PreferredFitStrategy *PreferredFitStrategy
+}
+```
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Test Plan
+
+- Unit and integration tests covering core changes.
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation, and anything particularly
+challenging to test, should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+### Graduation Criteria
+
+#### Alpha -> Beta Graduation
+
+- The KEP proposes an API change to scheduler component config to allow 
+  expressing an existing behavior in a different way. Since this configuration
+  is opt-in, it will not be guarded by a feature flag, and will graduate with
+  component config (i.e., will start in beta directly)
```

Comment:
I would be more explicit:

It would be disabled by default when using v1beta1 and enabled by default in v1beta2.